lua/lazy/manage/rocks.lua

```diff
@@ -0,0 +1,75 @@
+local Config = require("lazy.core.config")
+local Util = require("lazy.core.util")
+
+---@class LazyRock
+---@field plugin string
+---@field name string
+---@field spec string
+---@field installed boolean
+
+local M = {}
+---@type LazyRock[]
+M.rocks = {}
+
+---@param ... string
+---@return string[]
+function M.args(...)
+  local ret = { "--tree", Config.rocks.tree, "--lua-version", "5.1" }
+  vim.list_extend(ret, { ... })
+  return ret
+end
+
+---@param plugin LazyPlugin
+function M.get_rockspec(plugin)
+  assert(plugin.rocks and #plugin.rocks > 0, plugin.name .. " has no rocks")
+  local rockspec_file = Config.rocks.specs .. "/lazy-" .. plugin.name .. "-0.0-0.rockspec"
+  require("lazy.util").write_file(
+    rockspec_file,
+    ([[
+rockspec_format = "3.0"
+package = "lazy-%s"
+version = "0.0-0"
+source = { url = "%s" }
+dependencies = %s
```

Comment:
suggestion: some plugins have am `-scm-1.rockspec` in their root directory.
Maybe it would be feasible to automatically get the dependencies from there if a rockspec is present (and `rocks = true`?).

This would allow plugin developers to add/remove dependencies without it being a breaking change.

Reply:
> This would allow plugin developers to add/remove dependencies without it being a breaking change.

What I think is even more problematic is when there was a bug in a dependency which you can't fix on your side because users manage the dependencies themselves

Reply:
> (and rocks == true?).

regarding that I think it then would make sense to let rocks default to true
Otherwise it would be a breaking change for a plugin to switch to using a rockspec for dependencies